src/applications/facility-locator/containers/FacilitesMap.jsx

```diff
@@ -0,0 +1,469 @@
+
+    const setMobile = () => {
+      setIsMobile(window.innerWidth <= 481);
+    };
+
+    searchWithUrl();
+
+    const debouncedResize = vaDebounce(250, setMobile);
+    window.addEventListener('resize', debouncedResize);
+    return () => {
+      listener();
+      window.removeEventListener('resize', debouncedResize);
+    };
+  }, []);
+
+  const updateUrlParams = params => {
+    const { location, currentQuery } = props;
+
+    const queryParams = {
+      ...location.query,
+      zoomLevel: currentQuery.zoomLevel,
+      page: currentQuery.currentPage,
+      address: currentQuery.searchString,
+      facilityType: currentQuery.facilityType,
+      serviceType: currentQuery.serviceType,
+      ...params,
+    };
+
+    const queryStringObj = appendQuery(
+      `/find-locations${location.pathname}`,
+      queryParams,
+    );
+
+    browserHistory.push(queryStringObj);
+  };
+
+  useEffect(
+    () => {
+      if (isSearching) {
+        updateUrlParams({
+          location: `${props.currentQuery.position.latitude},${
+            props.currentQuery.position.longitude
+          }`,
+          context: props.currentQuery.context,
+          address: props.currentQuery.searchString,
+        });
+        const resultsPage = props.currentQuery.currentPage;
+
+        if (!props.searchBoundsInProgress) {
+          props.searchWithBounds({
+            bounds: props.currentQuery.bounds,
+            facilityType: props.currentQuery.facilityType,
+            serviceType: props.currentQuery.serviceType,
+            page: resultsPage,
+          });
+          setIsSearching(false);
+        }
+        if (searchResultTitleRef.current) {
+          setFocus(searchResultTitleRef.current);
+        }
+      }
+      return () => {
+        // remove the listeners
+      };
+    },
+    [props.currentQuery.id], // Handle search when query changes
+  );
+
+  const renderMarkers = locations => {
+    if (locations.length === 0) return;
+    const currentLocation = props.currentQuery.position;
+    const markersLetters = MARKER_LETTERS.values();
+    const sortedLocations = locations
+      .map(r => {
+        const distance = currentLocation
+          ? distBetween(
+              currentLocation.latitude,
+              currentLocation.longitude,
+              r.attributes.lat,
+              r.attributes.long,
+            )
+          : null;
+        return {
+          ...r,
+          distance,
+        };
+      })
+      .sort((resultA, resultB) => resultA.distance - resultB.distance);
+
+    sortedLocations.forEach(loc => {
+      const attrs = {
+        letter: markersLetters.next().value,
+      };
+      const markerElement = buildMarker('location', { loc, attrs });
+      new mapboxgl.Marker(markerElement)
+        .setLngLat([loc.attributes.long, loc.attributes.lat])
+        .addTo(map);
+    });
+
+    if (props.currentQuery.bounds) {
+      map.fitBounds(props.currentQuery.bounds); // {duration: 0} to disable animation
```

Comment:
I think we want to use the **markers** to determine the bounds, not the query.

Here is some (untested) code to try:

```js
const bounds = new mapboxgl.LngLatBounds();

locations.forEach(location => {
    bounds.extend(new mapboxgl.LngLat(location.latitude, location.longitude);
});

map.fitBounds(bounds);
```

Reply:
Actually you could probably compute the marker bounds in the same loop where we are adding the markers to the map.